pyinstl/instlMisc.py

```diff
@@ -111,38 +115,36 @@ def do_unwtar(self):
     def unwtar_a_file(self, wtar_file_path):
         try:
             wtar_folder_path, _ = os.path.split(wtar_file_path)
-            with tarfile.open(wtar_file_path, "r") as tar:
-                tar.extractall(wtar_folder_path)
+
+            if type(wtar_file_path) is list:
+                # a list means we should use multi read
+                with MultiFileReader("br", wtar_file_path) as fd:
```

Comment:
unwtar_a_file should be something like:
    def unwtar_a_file(self, *wtar_file_paths):
        try:
            wtar_folder_path, _ = os.path.split(wtar_file_paths[0])
            with utils.MultiFileReader("br", wtar_file_paths) as fd:
                with tarfile.open(fileobj=fd) as tar:
                    tar.extractall(wtar_folder_path)
            if self.no_artifacts:
                for wtar_file in wtar_file_paths:
                    os.remove(wtar_file)
            # self.dynamic_progress("Expanding {wtar_file_paths}".format(**locals()))
        except tarfile.TarError:
            print("tarfile error while opening file", os.path.abspath(wtar_file_paths))
            raise

so it can accept a single file or a list of files.


Inside the function do_unwtar, files_to_unwtar should be a list of lists.
If it's a .wtar file it will be  a list of a single item, otherwise join_split_files will be renamed find_split_files and will return a list of all .wtra.?? files belonging to the same archive.

So for example if a folder contains:
Resources.wtar
MacOS.wtar.aa
MacOS.wtar.ab

files_to_unwtar will be:
['Resources.wtar', ['MacOS.wtar.aa', 'MacOS.wtar.ab']]

Than for each item in files_to_unwtar you will call unwtar_a_file. e.g.:

                for wtar_file_paths in files_to_unwtar:
                    self.unwtar_a_file(*wtar_file_paths)